codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -365,14 +367,15 @@ public ChannelFuture close(final ChannelPromise promise) {
     }
 
     @Override
-    public void close(final ChannelHandlerContext ctx, final ChannelPromise promise) throws Exception {
+    public void close(final ChannelHandlerContext ctx, final ChannelOutboundInvokerCallback callback)
+            throws Exception {
         ChannelFuture f = finishEncode(ctx, ctx.newPromise());
-        f.addListener((ChannelFutureListener) f1 -> ctx.close(promise));
+        f.addListener((ChannelFutureListener) f1 -> ctx.close(callback));
```

Comment:
Do we really need to cast `ChannelFutureListener`?

Reply:
I think no.. but I this existed before the PR and I just do refactoring ;)